homeassistant/components/shelly/service.py

```diff
@@ -0,0 +1,36 @@
+"""Services for the Shelly integration."""
+from __future__ import annotations
+
+from homeassistant.const import ATTR_AREA_ID, ATTR_DEVICE_ID
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.service import async_extract_config_entry_ids
+
+from .const import ATTR_BETA, BLOCK, DATA_CONFIG_ENTRY, DOMAIN, RPC, SERVICE_OTA_UPDATE
+from .utils import get_device_entry_gen
+
+
+async def async_services_setup(hass: HomeAssistant) -> None:
+    """Set up services."""
+
+    async def async_service_ota_update(call: ServiceCall) -> None:
```

Comment:
Why is this registered as a service?
The button entity (used by this PR) was introduced to reduce these type of services

Reply:
> Those I need to rework this PR to implement the whole logic into the button and remove the custom service at all?

I don't see the need for a service at this point?

Reply:
makes absolutely sense, you convinced me 😃 
just for my "defense" when I started this PR, the button entity does not exists, therefore the service has been implemented first 🙈